File: Sorting/Bubble Sort.cpp
#include <vector>
#include <iostream>

using namespace std;

bool isSorted(vector<int>array){
  for (int i = 0; i < array.size()-1; i++){
    
    if(array[i] > array[i+1]) return false;
  }
  return true;
}

vector<int> bubbleSort(vector<int> array) {
  // Write your code here.

  if (isSorted(array) == 1) return array;
  
  int size = array.size();

  while(size > 1){
    
    for(int i = 0; i <size-1; i++){
      int temp = 0;

      if(array[i] > array[i+1]){
        temp = array[i];
        array[i] = array[i+1];
        array[i+1] = temp;
      }
    }
    size--;
  }
  
  return array;
}

File: Sorting/BubbleSort.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Module Module1

    Function isSorted(ByVal array As List(Of Integer)) As Boolean 
        For i As Integer = 0 To array.Count() - 2
            If array(i) > array(i+1) Then 
                Return False
            End If
        Next 
        
        Return True 
    End Function

    Function bubbleSort(ByVal array As List(Of Integer)) As List(Of Integer)

        If isSorted(array) Then 
            Return array
        End If

        Dim size = array.Count()

        While size > 1

            For i As Integer = 0 To size - 2
                If array(i) > array(i+1) Then
                    Dim temp = array(i)
                    array(i) = array(i+1)
                    array(i+1) = temp
                End If
            Next 
            size -= 1
        End While 

        Return array

    End Function

    Sub Main()
        Dim arr As List(Of Integer) = New List(Of Integer)({5, 3, 8, 6, 7, 2})

        Console.WriteLine("Unsorted array: ")

        For Each num As Integer In arr
            Console.Write(num & " ")
        Next

        Console.WriteLine()

        arr = bubbleSort(arr)

        Console.WriteLine("Sorted array: ")
        For Each num As Integer in arr
            Console.Write(num & " ")
        Next

        Console.Writeline()
        Console.ReadLine()
    End Sub
End Module